// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

enum AccesscontextmanagerServicePerimeterPerimeterTypeEnum {
  AccesscontextmanagerServicePerimeterPerimeterTypeEnumNO_VALUE_DO_NOT_USE = 0;
  AccesscontextmanagerServicePerimeterPerimeterTypeEnumPERIMETER_TYPE_REGULAR = 1;
  AccesscontextmanagerServicePerimeterPerimeterTypeEnumPERIMETER_TYPE_BRIDGE = 2;
}

message AccesscontextmanagerServicePerimeter {
  string title = 1;
  string description = 2;
  string create_time = 3;
  string update_time = 4;
  AccesscontextmanagerServicePerimeterPerimeterTypeEnum perimeter_type = 5;
  AccesscontextmanagerServicePerimeterStatus status = 6;
  string policy = 7;
  string name = 8;
  bool use_explicit_dry_run_spec = 9;
  AccesscontextmanagerServicePerimeterSpec spec = 10;
}

message AccesscontextmanagerServicePerimeterStatus {
  repeated string resources = 1;
  repeated string access_levels = 2;
  repeated string restricted_services = 3;
  AccesscontextmanagerServicePerimeterStatusVPCAccessibleServices vpc_accessible_services = 4;
}

message AccesscontextmanagerServicePerimeterStatusVPCAccessibleServices {
  bool enable_restriction = 1;
  repeated string allowed_services = 2;
}

message AccesscontextmanagerServicePerimeterSpec {
  repeated string resources = 1;
  repeated string access_levels = 2;
  repeated string restricted_services = 3;
  AccesscontextmanagerServicePerimeterSpecVPCAccessibleServices vpc_accessible_services = 4;
}

message AccesscontextmanagerServicePerimeterSpecVPCAccessibleServices {
  bool enable_restriction = 1;
  repeated string allowed_services = 2;
}

message ApplyAccesscontextmanagerServicePerimeterRequest {
  AccesscontextmanagerServicePerimeter resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteAccesscontextmanagerServicePerimeterRequest {
  string service_account_file = 1;
  AccesscontextmanagerServicePerimeter resource = 2;
}

message ListAccesscontextmanagerServicePerimeterRequest {
  string service_account_file = 1;
  string Policy = 2;
}

message ListAccesscontextmanagerServicePerimeterResponse {
  repeated AccesscontextmanagerServicePerimeter items = 1;
}

service AccesscontextmanagerServicePerimeterService {
  rpc ApplyAccesscontextmanagerServicePerimeter(ApplyAccesscontextmanagerServicePerimeterRequest) returns (AccesscontextmanagerServicePerimeter);
  rpc DeleteAccesscontextmanagerServicePerimeter(DeleteAccesscontextmanagerServicePerimeterRequest) returns (google.protobuf.Empty);
  rpc ListAccesscontextmanagerServicePerimeter(ListAccesscontextmanagerServicePerimeterRequest) returns (ListAccesscontextmanagerServicePerimeterResponse);
}
